AngularApp/src/app/modules/dashboard/tabs/tab-monitoring/tab-monitoring.component.ts

```diff
@@ -65,11 +67,38 @@ export class TabMonitoringComponent implements OnInit {
   }
 
   getMonitoringResponse() {
-    this.detectorResponse = null;
+    this.systemInvokerResponse = null;
     this.detectorId = this._route.parent.snapshot.params['detector'].toLowerCase();
     this._diagnosticApiService.getSystemInvoker(this.detectorId, this.systemInvokerId, this.dataSourceFlag, this.timeRangeInHours)
       .subscribe((response: DetectorResponse) => {
-        this.detectorResponse = response;
+        this.systemInvokerResponse = response;
+      }, (error: any) => {
+        this.error = error;
+      });
+  }
+
+  getDetectorResponse() {
+    this.detectorId = this._route.parent.snapshot.params['detector'].toLowerCase();
+
+    this._diagnosticApiService.getDetector(this.detectorId)
```

Comment:
One other option instead of getting the full detector data is to make GetDetectors call (which is much faster) and then search the response by detector Id. You'll get detector metadata in much faster way.

Reply:
Oh I see. Even for each detector I try to make a call to GetDetectors and then filter metadata of this detector, will still be faster right? Because GetDetectors call doesn't include invoker execution. 
I was concerned about the efficiency previously as well..